### ISSUE TITLE
Add support for https://lunrjs.com/ search index

### ISSUE THREAD

**Issue by randombenj** [docat-org/docat#527]
It would be nice to have search across projects, to support this let's have a look at lunrjs as it
seems to be the default frontend search thingy.

**Comment by goerks** [comment]
Hi Benj, I like this approach.

We are using the Hugo static website generator based on markdown documents. We use the [Relearn Hugo theme](https://mcshelby.github.io/hugo-theme-relearn/basics/customization/index.html#activate-search) which is shipped with the lunrjs search engine. Therefore each Hugo documentation has a `index.search.js` containing the prebuilt search index.

```javascript
var relearn_search_index = [
  {
    "content": " Documentation as Code You should write the documentation ...,
    "description": "Explains the principle of Documentation as Code.",
    "tags": null,
    "title": "Documentation as Code",
    "uri": "/training/advanced/docs-as-code/index.html"
  },
  ...
]
```

## Minimal setup

- Each project provides its own lunr search index (`index.search.js` in the document root directory)
- When searching in the docat search, the latest version of each project should be considered

## Advanced setup

- Allow to search in older versions too
- Support custom [scoring](https://lunrjs.com/guides/searching.html#scoring) and [boosts](https://lunrjs.com/guides/searching.html#boosts)

**Comment by randombenj** [comment]
@goerks We tried to implement search multiple times now, unfortunately it seems to be a harder problem than it looks ...

@reglim Is this something we looked into when you tried to implement search?

**Comment by reglim** [comment]
@randombenj Yes I did. I think this idea was dropped first of all because to serve the pre-built index it would need to be built in the backend which is not in JavaScript but Python. Also, Lunr doesn't allow you [update an existing index](https://github.com/olivernn/lunr.js/issues/284), which would mean that it would have to be rebuilt completely every time something is updated. 

An alternative could be [elasticlunr](https://github.com/weixsong/elasticlunr.js) that allows you to update documents, but that still has the issue of building it in the backend.

Also, in my testing, extracting all the text from the uploaded zip files can also be quite expensive.
